clients/ctl/admin-ui/src/features/data-qualifier/data-qualifier.slice.ts

```diff
@@ -64,19 +56,20 @@ export const {
   useDeleteDataQualifierMutation,
 } = dataQualifierApi;
 
+export interface State {}
+const initialState: State = {};
+
 export const dataQualifierSlice = createSlice({
   name: "dataQualifier",
   initialState,
-  reducers: {
-    setDataQualifiers: (state, action: PayloadAction<any>) => ({
-      ...state,
-      dataQualifier: action.payload,
-    }),
-  },
+  reducers: {},
 });
 
-export const { setDataQualifiers } = dataQualifierSlice.actions;
-export const selectDataQualifiers = (state: AppState) =>
-  state.dataQualifier.dataQualifier;
-
 export const { reducer } = dataQualifierSlice;
+
+const emptyDataQualifiers: DataQualifier[] = [];
```

Comment:
As a reminder: creating a constant for an empty selection results reduces some re-rendering by preserving the selector's memoization. Not important for performance at the moment, but it's a helpful habit.

Reply:
thanks for the explanation! did not know